lib/pb_extern.h

```diff
@@ -6,6 +6,10 @@
 /// `PUBNUB_EXTERN` before including any Pubnub header.
 
 #ifndef PUBNUB_EXTERN
+#ifdef __cplusplus
+#define PUBNUB_EXTERN extern "C"
+#else
 #define PUBNUB_EXTERN extern
+#endif
```

Comment:
For now its important that the CI passes. This change should fix things. I tested locally with both Make and CMake and using the simple external demo. All worked correctly.

```suggestion
#define PUBNUB_EXTERN extern
```

Reply:
Oops you get there before i did! Sorry i should have refreshed the page first :laughing:

Reply:
Bad timing :D

Reply:
Just noticed that too. Best closing the two open threads and moving onto the next problems! :rofl: 

Sorry for the confusion. Sometimes github is kind and updates threads automatically... sometimes it doesn't

Reply:
Okay. I removed the extern.
I forgot that we wrap the whole C code base into `extern "C" {}` block. :/